engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnEventSubscriptionBehavior.java

```diff
@@ -263,6 +264,24 @@ public void triggerEventBasedGateway(final BpmnElementContext context) {
         eventInstanceKey, ProcessInstanceIntent.COMPLETE_ELEMENT, elementRecord);
   }
 
+  private void publishTriggeredEvent(
```

Comment:
What is the difference between `publishTriggeredBoundaryEvent` and `activateTriggeredEvent`? Aren't these two ways to do the same thing?

Reply:
Yes. `publishTriggeredBoundaryEvent()` will be replaced by `activateTriggeredEvent()` for migrated element processors. 

`publishTriggeredBoundaryEvent()` is based on deferred records and the usage of `event_occurred`. When all processors are migrated then we can remove it.